bishop.cc

```diff
@@ -64,11 +74,14 @@ void Bishop::updateLegalMoves() {
         if (((*(this->getBoard()))[x + i][y + i])) {
             if(isLegal(move1)) {
                 legalMoves.push_back(move1);
+                moveStates.push_back(1);
             }
             break;
         }
         legalMoves.push_back(move1);
+        moveStates.push_back(0);
     }
 
     this->getLegalMoves() = &legalMoves;
```

Comment:
I'm a little confused about this part of the code. I haven't yet seen 'this' being used with a method, I've only seen it being used with fields so far. getLegalMoves() returns legalMoves (not the vector in this method, the field) by definition. I'm not sure what setting the method to point to the new legalMoves does.

Also, if I'm not wrong, legalMoves defined within this method will be deleted once the function ends i.e. the vector goes out of scope so the address is invalid anyway, no? I have the same concern with moveStates.